evaluation/tasks/hans/hans.py

```diff
@@ -0,0 +1,138 @@
+        # Reference for metrics: https://github.com/tommccoy1/hans/blob/master/evaluate_heur_output.py
+        dataset = HANSDataset(self.tokenizer)
+        accuracy = 0.0
+        heuristic_ent_correct_count_dict = {}
+        subcase_correct_count_dict = {}
+        template_correct_count_dict = {}
+        heuristic_ent_incorrect_count_dict = {}
+        subcase_incorrect_count_dict = {}
+        template_incorrect_count_dict = {}
+        heuristic_nonent_correct_count_dict = {}
+        heuristic_nonent_incorrect_count_dict = {}
+
+        for heuristic in dataset.heuristics:
+            heuristic_ent_correct_count_dict[heuristic] = 0.0
+            heuristic_ent_incorrect_count_dict[heuristic] = 0.0
+            heuristic_nonent_correct_count_dict[heuristic] = 0.0
+            heuristic_nonent_incorrect_count_dict[heuristic] = 0.0
+        for subcase in dataset.subcases:
+            subcase_correct_count_dict[subcase] = 0.0
+            subcase_incorrect_count_dict[subcase] = 0.0
+        for template in dataset.templates:
+            template_correct_count_dict[template] = 0.0
+            template_incorrect_count_dict[template] = 0.0
+
+        for item in tqdm(dataset, desc=f"Evaluating {self.get_display_name()}"):
+            output = self.model.generate(
+                input_ids=item["input_ids"].to(self.device),
+                attention_mask=item["attention_mask"].to(self.device),
+                max_length=min(item["input_len"] * 2, self.model.config.n_positions),
+            )
+            prompt_len = len(item["prompt"])
+            decoded_output = self.tokenizer.decode(output[0], skip_special_tokens=True)
+            predicted_answer = decoded_output[prompt_len:].lower()
+            correct_label = ["true", "false"][item["label"]]
+            if correct_label in predicted_answer:
+                accuracy += 1.0
+                if correct_label == "true":
+                    heuristic_ent_correct_count_dict[item["heuristic"]] += 1
+                else:
+                    heuristic_nonent_correct_count_dict[item["heuristic"]] += 1
+                subcase_correct_count_dict[item["subcase"]] += 1
+                template_correct_count_dict[item["template"]] += 1
+            else:
+                if correct_label == "true":
+                    heuristic_ent_incorrect_count_dict[item["heuristic"]] += 1
+                else:
+                    heuristic_nonent_incorrect_count_dict[item["heuristic"]] += 1
+                subcase_incorrect_count_dict[item["subcase"]] += 1
+                template_incorrect_count_dict[item["template"]] += 1
+
+        self.metrics["hans_overall_accuracy"] = accuracy / len(dataset) * 100
+        for heuristic in dataset.heuristics:
+            total = heuristic_ent_correct_count_dict[heuristic] + heuristic_ent_incorrect_count_dict[heuristic]
+            self.metrics["hans_{}_entailed_accuracy".format(heuristic)] = heuristic_ent_correct_count_dict[heuristic] / total * 100
```

Comment:
nit: The logic to calculate accuracy is repeated a few times in this function. I think it'd be useful to create a helper method for it.